workflows/index-generation/generate_lineage_csvs.py

```diff
@@ -184,24 +206,26 @@ def version_taxon_lineages(
     scheme described above
     """
     previous_lineages = {}
+    previous_lineages_version = None
     if previous_lineages_filename:
         with gzip.open(previous_lineages_filename, "rt") as f:
             for row in csv.DictReader(f):
-                previous_lineages[row["taxid"]] = row
+                previous_lineages[(row["taxid"], row["version_end"])] = row
+                if (
+                    not previous_lineages_version
+                    or previous_lineages_version < row["version_end"]
```

Comment:
pretty cool that the string comparison works well with these operators, didn't know python could do that!